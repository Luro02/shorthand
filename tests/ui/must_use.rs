#![deny(unused_must_use)]
use shorthand::ShortHand;

#[derive(Default, ShortHand)]
#[shorthand(enable(must_use))]
struct Example {
    value: String,
}

fn test_must_use_fail() {
    // should fail because it is marked as unused
    Example::default().value();
}

fn test_should_be_fine() {
    Example::default().set_value("x".to_string());
}

fn main() {}
