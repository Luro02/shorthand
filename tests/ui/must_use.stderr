error: unused return value of `Example::value` that must be used
  --> $DIR/must_use.rs:12:5
   |
12 |     Example::default().value();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/must_use.rs:1:9
   |
1  | #![deny(unused_must_use)]
   |         ^^^^^^^^^^^^^^^
